Add initialize action and basic test

duncanmortimer opened the pull request:
Currently, there isn't an explicit way to customize the initial value of the context when handling a request, leading to ad-hoc methods (such as using the 'service-available?' decision point).

This pull request adds an 'initialize' action prior to 'service-available?' which provides this option.

I'll also add a pull request to the gh-pages branch with updated documentation.

ordnungswidrig commented on test/test_flow.clj:
```diff
@@ -5,7 +5,10 @@
-
+(facts "customize the initial context"
+       (let [resp ((resource :initialize {::field "some initial context"}
+                             :handle-ok (fn [ctx] (::field ctx))) (request :get "/"))]
```
`(fn [ctx] (::field ctx))` can be simplified to `::field`.

ordnungswidrig commented:
I like the idea. I'm not too sure about the name of the action. "Initialize" seems obvious but I wonder if there's something better.

duncanmortimer commented:
Glad you like the idea.

WRT the name, I guess 'Initialize' is quite generic.  Not sure I have any better suggestions..  ... maybe 'prepare'?  Any thoughts?

duncanmortimer commented:
Hi @ordnungswidrig 

Anything else you'd like me to do before this can be merged?
I'll update the gh-pages as well, assuming this is fine.

Cheers, Duncan

duncanmortimer commented:
Hi @ordnungswidrig - any other comments?  Would 'prepare-context' be a better name?

I'll update the gh-pages, just want to verify that there isn't anything else needed here first.

Cheers!

ordnungswidrig commented on src/liberator/core.clj:
```diff
@@ -575,10 +579,9 @@
-                         (map-values make-function (merge default-functions kvs))
-                         :representation {}})
+    (prepare {:request request
+                 :resource (map-values make-function (merge default-functions kvs))
```
indentation...

ordnungswidrig commented:
Hey @duncanmortimer, sorry for being silent for so long. I'm not too happy with `prepare`, this somehow makes me think of side effects ("prepare transaction"). What about `initialize-context`?

duncanmortimer commented:
Hi @ordnungswidrig - No problems.  Life's busy! :-)  I've just updated to :initialize-context, and fixed the indentation issues you've noted (also updated gh-pages -- will comment there in a sec).  Hopefully good now?

ordnungswidrig commented:
Great! Would you mind to squash the commits together? The back-and-forth on the naming does not need to be part of the history :-) I will merge after that.

duncanmortimer commented:
Done.

ordnungswidrig commented:
Thanks for the patch!

duncanmortimer commented:
Great! Cheers.
